import Numpy as np

In [17]:
import numpy as np
import pandas as pd


In [18]:

# Set seed for reproducibility
np.random.seed(42)

# Define dataset size (60 samples, falling in the 50-100 range)
n_samples = 60

# Generate synthetic features
cgpa = np.random.uniform(6.0, 10.0, n_samples).round(2)
study_hours = np.random.normal(5.0, 1.5, n_samples).clip(1.0, 10.0).round(1)
aptitude_score = np.random.randint(50, 100, n_samples)

# Generate a synthetic binary target (placement status: 0 or 1) influenced by CGPA and study hours
score_metric = (cgpa - 6.0) * 2 + (study_hours * 0.5)
probability = 1 / (1 + np.exp(-(score_metric - 4)))
placement_status = (np.random.random(n_samples) < probability).astype(int)

# Combine all arrays column-wise into a single 2D NumPy dataset
# Columns: [CGPA, Study_Hours, Aptitude_Score, Placement_Status]
dataset = np.column_stack((cgpa, study_hours, aptitude_score, placement_status))

# Display dataset details
print("Dataset Shape:", dataset.shape)
print("\nFirst 5 rows [CGPA, Study Hours, Aptitude Score, Placement Status]:")
print(dataset[:5])
df = pd.DataFrame(dataset, columns=['cgpa', 'study_hours', 'aptitude_score', 'placement_status'])

# First 5 rows check karne ke liye
print(df.head())

# DataFrame ka basic structure aur data types dekhne ke liye
print("\nDataFrame Info:")
print(df.info())

Dataset Shape: (60, 4)

First 5 rows [CGPA, Study Hours, Aptitude Score, Placement Status]:
[[ 7.5   2.8  64.    1.  ]
 [ 9.8   3.9  81.    1.  ]
 [ 8.93  4.3  81.    1.  ]
 [ 8.39  6.6  73.    1.  ]
 [ 6.62  5.5  90.    0.  ]]
   cgpa  study_hours  aptitude_score  placement_status
0  7.50          2.8            64.0               1.0
1  9.80          3.9            81.0               1.0
2  8.93          4.3            81.0               1.0
3  8.39          6.6            73.0               1.0
4  6.62          5.5            90.0               0.0

DataFrame Info:
<class 'pandas.DataFrame'>
RangeIndex: 60 entries, 0 to 59
Data columns (total 4 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   cgpa              60 non-null     float64
 1   study_hours       60 non-null     float64
 2   aptitude_score    60 non-null     float64
 3   placement_status  60 non-null     float64
dtypes: float64(4)
memory usage: 2.0 KB
None


In [19]:
df

,cgpa,study_hours,aptitude_score,placement_status
0,7.50,2.8,64.0,1.0
1,9.80,3.9,81.0,1.0
2,8.93,4.3,81.0,1.0
3,8.39,6.6,73.0,1.0
4,6.62,5.5,90.0,0.0
5,6.62,2.4,98.0,0.0
6,6.23,5.5,98.0,0.0
7,9.46,4.4,61.0,1.0
8,8.40,4.0,88.0,1.0
9,8.83,5.9,51.0,1.0


In [20]:
before_outliers = df['cgpa'].mean()
before_outliers

np.float64(7.870000000000001)

In [24]:
df.loc[5,'cgpa'] = 25
df.loc[25,'cgpa'] = -24
df.loc[52,'cgpa'] = 23
df.loc[43,'cgpa'] = 30
df.loc[43,'cgpa'] = 20
df.loc[33,'cgpa'] = -20
df.loc[13,'cgpa'] = 0



In [25]:
after_outliers = df['cgpa'].mean()
after_outliers


np.float64(7.4065)

In [26]:
print("Here is the difference betweeen before and after outliers")
after_outliers-before_outliers 

Here is the difference betweeen before and after outliers


np.float64(-0.4635000000000007)

In they are the difference between the before and after outliers are -0.46 approxmatily